front/components/agent_builder/capabilities/knowledge/KnowledgeConfigurationSheet.tsx

```diff
@@ -81,9 +81,20 @@ export function KnowledgeConfigurationSheet({
     return CAPABILITY_CONFIGS["search"];
   });
 
-  const handleClose = () => {
-    onClose();
-  };
+  // Custom open hook to only have debounce when we close.
+  // We use this value to unmount the Sheet Content, and we need
+  // debounce when closing to avoid messing up the closing animation.
+  // 300ms is vibe based.
```

Comment:
I've fixed this by keeping the component mounted but controlling its state manually from the parents, you should be able to get rid of this 🤓